<?xml version="1.0" ?>
<launch>

  <arg name="camera_id"			default="2018-09-017-LC3"/>
  <arg name="camera_name"		default="phoxi"/>
  <arg name="marker_map_dir"		default="$(dirname)/../config"/>
  <arg name="marker_map"		default="markermap-100-101-102-103"/>
  <arg name="marker_frame"		default="marker_frame"/>
  <arg name="planarity_tolerance"	default="0.001"/>
  <arg name="enable_pointcloud"		default="false"/>
  <arg name="manager"			default=""/>
  <arg name="vis"			default="true"/>

  <arg name="camera_frame"		value="$(arg camera_name)_sensor"/>
  <node name="static_transform_publisher"
	pkg="tf" type="static_transform_publisher"
        args="0 0 0 0 0 0 base_link $(arg camera_frame) 10"/>

  <include file="$(find aist_phoxi_camera)/launch/run.launch">
    <arg name="id"			value="$(arg camera_id)"/>
    <arg name="camera_name"		value="$(arg camera_name)"/>
    <arg name="frame"			value="$(arg camera_frame)"/>
    <arg name="trigger_mode"		value="0"/>
    <arg name="point_format"		value="1"/>	<!-- with RGB -->
    <arg name="send_point_cloud"	value="$(arg enable_pointcloud)"/>
    <arg name="manager"			value="$(arg manager)"/>
  </include>

  <include file="$(dirname)/run.launch">
    <arg name="camera_info_topic"	value="$(arg camera_name)/camera_info"/>
    <arg name="image_topic"		value="$(arg camera_name)/texture"/>
    <arg name="depth_topic"		value="$(arg camera_name)/depth_map"/>
    <arg if="$(arg enable_pointcloud)"
	 name="cloud_topic"		value="$(arg camera_name)/pointcloud"/>
    <arg name="marker_map_dir"		value="$(arg marker_map_dir)"/>
    <arg name="marker_map"		value="$(arg marker_map)"/>
    <arg name="marker_frame"		value="$(arg marker_frame)"/>
    <arg name="planarity_tolerance"	value="$(arg planarity_tolerance)"/>
    <arg name="manager"			value="$(arg manager)"/>
  </include>

  <group if="$(arg vis)">
    <node name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
	  args="-d $(dirname)/phoxi.rviz"/>

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
  </group>

</launch>
